Src/VimCore/CaretRegisterMap.fs

```diff
@@ -0,0 +1,123 @@
+﻿namespace Vim
+
+open System.Collections.Generic
+
+/// Mutable caret index object
+type CaretIndex() =
+    let mutable _caretIndex = 0
+    member x.Value
+        with get() = _caretIndex
+        and set value = _caretIndex <- value
+
+/// Caret-aware IRegisterValueBacking implementation for the unnamed register.
+/// If there are multiple carets, each caret gets its own distinct value
+type CaretUnnamedRegisterValueBacking
+    (
+        _caretIndex: CaretIndex,
+        _unnamedRegister: Register
+    ) =
+
+    let valueMap = Dictionary<int, RegisterValue>()

```

Comment:
Won't the caret index values be contiguous? If so then why a map instead of a list?

Reply:
This might possible if the caret register map was aware of how many carets there were. Let me think about this.

Reply:
I think it's better to think of this as an associative array where items come into existence when you refer to them.